examples/7guis/cells.rs

```diff
@@ -0,0 +1,245 @@
+/// So when node A is updated, node B also need to re updated.
+#[derive(Default)]
+struct DepGraph<A, B> {
+    index1: HashMap<A, HashSet<B>>,
+    index2: HashMap<B, HashSet<A>>,
+}
+
+impl<A: Clone + Eq + Hash, B: Clone + Eq + Hash> DepGraph<A, B> {
+    pub fn add_dep(&mut self, a: A, b: B) {
+        self.index1.entry(a.clone()).or_default().insert(b.clone());
+        self.index2.entry(b).or_default().insert(a);
+    }
+
+    pub fn deps<'a>(&'a self, a: &A) -> impl Iterator<Item = &B> + 'a {
+        self.index1.get(a).into_iter().flat_map(|x| x.iter())
+    }
+
+    pub fn remove_rev_deps(&mut self, b: &B) {
+        if let Some(h) = self.index2.remove(b) {
+            for a in h {
+                self.index1.get_mut(&a).map(|x| x.remove(b));
+            }
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum Expr {
+    Value(f32),
+    Cell(usize, usize),
+    Add(Box<(Expr, Expr)>),
+    Sub(Box<(Expr, Expr)>),
+}
+
+fn parse_formula(formula: &str) -> Option<Expr> {
+    let formula = formula.trim();
+    let alpha_n = formula.find(|c: char| !c.is_ascii_alphabetic()).unwrap_or(formula.len());
+    let num_n = formula[alpha_n..]
+        .find(|c: char| !c.is_ascii_digit() && c != '.')
+        .map_or(formula.len(), |x| x + alpha_n);
+
+    let e = if alpha_n > 0 {
+        if alpha_n != 1 {
+            return None;
+        };
+        let col = formula.as_bytes()[0].to_ascii_lowercase() - b'a';
+        let row = formula[alpha_n..num_n].parse().ok()?;
+        Expr::Cell(row, col as usize)
+    } else if num_n > 0 {
+        Expr::Value(formula[alpha_n..num_n].parse().ok()?)
+    } else {
+        return None;
+    };
+
+    let rest = formula[num_n..].trim();
+
+    if rest.is_empty() {
+        Some(e)
+    } else if let Some(x) = rest.strip_prefix("+") {
+        Some(Expr::Add(Box::new((e, parse_formula(x)?))))
+    } else if let Some(x) = rest.strip_prefix("-") {
+        Some(Expr::Sub(Box::new((e, parse_formula(x)?))))
+    } else {
+        None
+    }
+}
+
+#[test]
+fn test_parse_formula() {
+    assert_eq!(parse_formula("42"), Some(Expr::Value(42.0)));
+    assert_eq!(parse_formula(" 49.5 "), Some(Expr::Value(49.5)));
+    assert_eq!(parse_formula("B4"), Some(Expr::Cell(4, 1)));
+    assert_eq!(parse_formula("B 4"), None);
+    assert_eq!(parse_formula("B4.2"), None);
+    assert_eq!(parse_formula("AB5"), None);
+    assert_eq!(parse_formula("4B"), None);
+    assert_eq!(
+        parse_formula("8 + C6"),
+        Some(Expr::Add(Box::new((Expr::Value(8.), Expr::Cell(6, 2)))))
+    );
+    assert_eq!(
+        parse_formula(" a9-b2 "),
+        Some(Expr::Sub(Box::new((Expr::Cell(9, 0), Expr::Cell(2, 1)))))
+    );
+    assert_eq!(
+        parse_formula("D22+B12+85.5"),
+        Some(Expr::Add(Box::new((
+            Expr::Cell(22, 3),
+            Expr::Add(Box::new((Expr::Cell(12, 1), Expr::Value(85.5))))
+        ))))
+    );
+}
+
+struct RowModel {
+    row: usize,
+    row_data: RefCell<Vec<CellContent>>,
```

Comment:
`row_data` is already used in Model. But I think it is fine here.
We should instead replace the `row` term in Model with something like element, because it is really confusing in nested models.
But that will be breaking.